tests/Iterator/GlobIteratorErrorTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the webmozart/glob package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Webmozart\Glob\Tests\Iterator;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Webmozart\Glob\Iterator\GlobIterator;
+use Webmozart\Glob\Test\TestUtil;
+
+/**
+ * @since  1.0
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ */
+class GlobIteratorErrorTest extends \PHPUnit\Framework\TestCase
+{
+    private $tempDir;
+
+    protected function setUp(): void
+    {
+        $this->tempDir = TestUtil::makeTempDir('webmozart-glob', __CLASS__);
+
+        $filesystem = new Filesystem();
+        $filesystem->mirror(__DIR__.'/../Fixtures', $this->tempDir);
+    }
+
+    protected function tearDown(): void
+    {
+        $filesystem = new Filesystem();
+        chmod($this->tempDir.'/js', 0777);
+        $filesystem->remove($this->tempDir);
+    }
+
+    public function testIterateError()
+    {
+        if (defined('PHP_WINDOWS_VERSION_MAJOR')) {
+            $this->markTestSkipped('chmod tests only on Linux.');
+
+            return;
+        }
+        $this->expectException(\UnexpectedValueException::class);
```

Comment:
Let's move this expectation right before the statement that is supposed to throw

Reply:
I made a mistake in the arguments... I combined flags with mode... I must really need some sleep :D

Reply:
Hm, all tests pass here, but I noticed the mistake when I included the code in vscode-php-intellisense ...
Can you make the workflow run on this repo?

Reply:
Ah thanks. Will look at the tests again shortly... I'll try to run them in a VM instead of WSL.

Reply:
I think its failing because some checks run under root (alpine docker?). Added logic to skip test in that case. Please re-run.

Reply:
All tests are OK now. I checked that they actually get executed where it matters. Linux and macos - where the current user isn't root.